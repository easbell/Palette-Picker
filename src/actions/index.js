export const isLoading = (bool) => ({
  type: 'IS_LOADING',
  isLoading: bool
});

export const hasErrored = (message) => ({
  type: 'HAS_ERRORED',
  message
});

export const setProjects = (projects) => ({
  type: 'SET_PROJECTS',
  projects
});

export const addProject = (project) => ({
  type: 'ADD_PROJECT',
  project
});

export const addPalette = (palette) => ({
  type: 'ADD_PALETTE',
  palette
});

export const setPalettes = (palettes) => ({
  type: 'SET_PALETTES',
  palettes
});