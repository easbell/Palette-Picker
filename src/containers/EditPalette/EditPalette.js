import React from 'react'
import { connect } from 'react-redux'
import '../../components/DisplayPalette/DisplayPalette.css'
import { setPalettes } from '../../actions'
import { deleteFromBE } from '../../helpers/apiCalls'

export const EditPalette = (props) => {
  const { id, palette_name, color_1, color_2, color_3, color_4, color_5 } = props.palette
  const { palettes } = props

  const deletePalette = (id) => {
    const updatedPalettes = palettes.filter(palette => { 
      return palette.id !== id
    })
    props.setPalettes(updatedPalettes);
    const url = process.env.REACT_APP_BACKEND_URL + `api/v1/palettes/${id}`
    deleteFromBE(url, id)
  }

  return(
    <div className='edit-palette-card'>
      <div className='palette-card edit-palette'>
        <h3 className='palette-name'>
          {palette_name}
        </h3>
        <div className='display-colors'>
          <div style={{backgroundColor: color_1}} className='display-color'></div>
          <div style={{backgroundColor: color_2}} className='display-color'></div>
          <div style={{backgroundColor: color_3}} className='display-color'></div>
          <div style={{backgroundColor: color_4}} className='display-color'></div>
          <div style={{backgroundColor: color_5}} className='display-color'></div>
        </div>
      </div>
      <div className='delete-palette' onClick={() => deletePalette(id)}><i className="far fa-trash-alt"></i></div>
    </div>
  )
}

const mapStateToProps = (state) => ({
  palettes: state.palettes,
})

const mapDispatchToProps = (dispatch) => ({
  setPalettes: (palettes) => dispatch(setPalettes(palettes))
})

export default connect(mapStateToProps, mapDispatchToProps)(EditPalette)

