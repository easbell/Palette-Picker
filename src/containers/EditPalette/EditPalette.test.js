import React from 'react'
import { EditPalette, mapStateToProps, mapDispatchToProps } from './EditPalette'
import { shallow } from 'enzyme'
import { setPalettes } from '../../actions'

describe('EditPalette', () => {
  let wrapper
  let mockPalette = {
    id: 1,
    palette_name: 'my palette',
    color_1: 'blue',
    color_2: 'pink',
    color_3: 'yellow',
    color_4: 'green',
    color_5: 'red'
  }

  it('should match the snapshot', () => {
    wrapper = shallow(<EditPalette palette={mockPalette} />)
    expect(wrapper).toMatchSnapshot()
  })

  describe('deletePalette', () => {

    it.skip('should dispatch setPalettes with the updated palettes', () => {
      wrapper = shallow(<EditPalette 
        palette={mockPalette} 
        setPalettes={jest.fn()}
        palettes = {[{id: 1}, {id: 2}]}
        />
      )

      wrapper.find('.delete-palette').simulate('click')
      
      expect(wrapper.props.setPalettes).toHaveBeenCalled()
    })
  })

  describe('mapStateToProps', () => {
    it('returns correct props', () => {
      const mockState = {
        palettes: ['palettes']
      }
      const expectedProps = {
        palettes: ['palettes']
      }

      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(expectedProps)
    });
  });

  describe('mapDispatchToProps', () => {
    it('calls dispatch with a setPaletts action', () => {
      const mockDispatch = jest.fn()
      const mockPalettes = ['palettes']
      const actionToDispatch = setPalettes(mockPalettes)

      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.setPalettes(mockPalettes)

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })
  })
})