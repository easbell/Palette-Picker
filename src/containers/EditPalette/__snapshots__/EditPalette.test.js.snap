// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditPalette should match the snapshot 1`] = `
<div
  className="edit-palette-card"
>
  <div
    className="palette-card edit-palette"
  >
    <h3
      className="palette-name"
    >
      my palette
    </h3>
    <div
      className="display-colors"
    >
      <div
        className="display-color"
        style={
          Object {
            "backgroundColor": "blue",
          }
        }
      />
      <div
        className="display-color"
        style={
          Object {
            "backgroundColor": "pink",
          }
        }
      />
      <div
        className="display-color"
        style={
          Object {
            "backgroundColor": "yellow",
          }
        }
      />
      <div
        className="display-color"
        style={
          Object {
            "backgroundColor": "green",
          }
        }
      />
      <div
        className="display-color"
        style={
          Object {
            "backgroundColor": "red",
          }
        }
      />
    </div>
  </div>
  <div
    className="delete-palette"
    onClick={[Function]}
  >
    <i
      className="far fa-trash-alt"
    />
  </div>
</div>
`;
